aten/src/ATen/core/op_registration/op_registration.cpp

```diff
@@ -140,45 +158,55 @@ Library& Library::_def(c10::either<OperatorName, FunctionSchema>&& name_or_schem
       OperatorName name = std::move(name_or_schema).left();
       FunctionSchema s = f.schema_->cloneWithName(std::move(name.name), std::move(name.overload_name));
       s.setAliasAnalysis(c10::AliasAnalysisKind::CONSERVATIVE);
       return s;
     }
   }();
-  if (ns_.has_value()) schema.setNamespaceIfNotSet(ns_->c_str());
-  TORCH_CHECK(!(f.dispatch_key_.has_value() && dispatch_key_.has_value()), "Cannot specify a different dispatch key inside a TORCH_LIBRARY_IMPL; please declare a separate TORCH_LIBRARY_IMPL for your dispatch key");
+  if (ns_.has_value()) {
+    TORCH_CHECK(schema.setNamespaceIfNotSet(ns_->c_str()), "Attempted to def ", toString(schema.operator_name()), " which is explicitly qualified with a namespace inside a TORCH_LIBRARY, which is not allowed.  If TORCH_LIBRARY's namespace matches the explicitly given namespace, remove the qualifier; otherwise, please move your def into a TORCH_LIBRARY block with the correct namespace, or give it a different namespace.  Registration site was ", file_, ":", line_);
+  }
+  TORCH_CHECK(!(f.dispatch_key_.has_value() && dispatch_key_.has_value()), "Cannot specify a different dispatch key inside a TORCH_LIBRARY_IMPL; please declare a separate TORCH_LIBRARY_IMPL for your dispatch key.  Registration site was ", file_, ":", line_);
   auto dispatch_key = f.dispatch_key_.has_value() ? f.dispatch_key_ : dispatch_key_;
   // Retain the OperatorName for Impl call
   OperatorName name = schema.operator_name();
-  registrars_.emplace_back(Dispatcher::singleton().registerDef(std::move(schema)));
-  registrars_.emplace_back(Dispatcher::singleton().registerImpl(name, dispatch_key, std::move(f.func_), std::move(f.schema_), std::move(f.debug_)));
+  registrars_.emplace_back(Dispatcher::singleton().registerDef(std::move(schema), debugString("", file_, line_)));
+  registrars_.emplace_back(Dispatcher::singleton().registerImpl(name, dispatch_key, std::move(f.func_), std::move(f.schema_), debugString(std::move(f.debug_), file_, line_)));
   return *this;
 }
 
 Library& Library::_impl(const char* name_str, CppFunction&& f) & {
   auto name = torch::jit::parseName(name_str);
-  if (ns_.has_value()) name.setNamespaceIfNotSet(ns_->c_str());
-  TORCH_CHECK(!(f.dispatch_key_.has_value() && dispatch_key_.has_value()), "Cannot specify a different dispatch key inside a TORCH_LIBRARY_IMPL; please declare a separate TORCH_LIBRARY_IMPL for your dispatch key");
+  if (ns_.has_value()) {
+    TORCH_CHECK(name.setNamespaceIfNotSet(ns_->c_str()), "Attempted to impl ", toString(name), " which is explicitly qualified with a namespace inside a TORCH_LIBRARY, which is not allowed.  If TORCH_LIBRARY's namespace matches the explicitly given namespace, remove the qualifier; otherwise, please place it in an separate TORCH_LIBRARY_IMPL block to make it clear that you are overriding behavior for an operator in a different library.  Registration site was ", file_, ":", line_);
```

Comment:
WDYT about wrapping these messages? I guess the idea is to keep them out of the way of the code, but they can be informative and the side scrolling gets pretty crazy. YMMV I guess.

Reply:
At this point it will be kind of annoying to adjust the formatting in the stack diff because of merge conflicts. I could do this after though.